base/db/R/get.trait.data.R

```diff
@@ -42,6 +43,45 @@ get.trait.data <-
     PEcAn.logger::logger.severe('At least one pft in settings is missing its "outdir"')
   }
   
+  #check for flatfile path if present use it 
+  file_path <- input_file 
+  if(is.null(file_path)){
+    file.path <- pfts$file_path
+  }
+  use_flatfile <- !is.null(file_path) && file.exists(file_path)
+  
+  if (use_flatfile) {
```

Comment:
Possible implementation of the error-if-not-exist pattern (If you like this, need to also add `@importFrom rlang %||%` in the Roxygen block above):
```suggestion
  #check for flatfile path, if present use it 
  file_path <- input_file %||% pfts$file_path
  if (!is.null(file_path) {
    if (!file.exists(file_path) {
      PEcAn.logger::logger.error("trait data file not found at specified path", sQuote(file_path))
    }
```